dipy/workflows/base.py

```diff
@@ -0,0 +1,129 @@
+from dipy.fixes import argparse as arg
+import inspect
+
+
+try:
```

Comment:
You might want to use the `optional_package` machinery here. For an example, see: https://github.com/nipy/dipy/blob/master/dipy/viz/fvtk.py#L32